consult.el

```diff
@@ -1312,5 +1317,40 @@ Prepend PREFIX in front of all items."
 
 (add-hook 'consult-preview-mode-hook #'consult--icomplete-preview-setup)
 
+(defun consult--flymake-candidates ()
+  "Return Flymake errors as alist."
+  (consult--forbid-minibuffer)
+  (let* ((diagnostics (flymake-diagnostics))
+	 (buffer-width (apply #'max (mapcar (lambda (x) (length (format "%s" (flymake-diagnostic-buffer x)))) diagnostics)))
+	 (category-width (apply #'max (mapcar (lambda (x) (length (symbol-name (flymake-diagnostic-type x)))) diagnostics)))
+         (fmt (format "%%-%ds %%-%ds %%s" buffer-width category-width)))
+    (mapcar
+     (lambda (diag)
+       (with-current-buffer (flymake--diag-buffer diag)
+	 (goto-char (flymake--diag-beg diag)))
+       (cons
+	(format fmt
+		(flymake-diagnostic-buffer diag)
+		(propertize
+		 (symbol-name (flymake-diagnostic-type diag))
+		 'face
+		 (flymake--lookup-type-property (flymake-diagnostic-type diag) 'mode-line-face))
+		(flymake-diagnostic-text diag))
+	(point-marker)))
+     (flymake-diagnostics))))
+
+;;;###autoload
+(defun consult-flymake ()
+  "Jump to Flymake diagnostic."
+  (interactive)
+  (consult--jump
+   (consult--read "Flymake diagnostic: "
+                  (consult--with-increased-gc (consult--flymake-candidates))
```

Comment:
Please add narrowing support via :narrow and consult--narrow-candidate. See consult-flycheck.

Reply:
Done.